src/bridges/compound/CompoundBridge.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright 2022 Spilsbury Holdings Ltd
+pragma solidity >=0.8.0 <=0.8.10;
+pragma abicoder v2;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IDefiBridge} from "../../interfaces/IDefiBridge.sol";
+import {AztecTypes} from "../../aztec/AztecTypes.sol";
+import "../../interfaces/IRollupProcessor.sol";
+import "./interfaces/ICERC20.sol";
+import "./interfaces/ICETH.sol";
+
+/**
+ * @title Aztec Connect Bridge for Compound protocol
+ * @notice You can use this contract to mint or redeem cTokens.
+ * @dev Implementation of the IDefiBridge interface for cTokens.
+ */
+contract CompoundBridge is IDefiBridge {
+    using SafeERC20 for IERC20;
+
+    address public immutable rollupProcessor;
+
+    error InvalidCaller();
+    error IncorrectInputAsset();
+    error IncorrectOutputAsset();
+    error IncorrectAuxData();
+    error AsyncModeDisabled();
+
+    constructor(address _rollupProcessor) {
+        rollupProcessor = _rollupProcessor;
+    }
+
+    receive() external payable {}
+
+    /**
+     * @notice Function which mints and burns cTokens in an exchange for the underlying asset.
+     * @dev This method can only be called from RollupProcessor.sol. If auxData is 0 the mint flow is executed,
+     * if 1 redeem flow.
+     *
+     * @param inputAssetA - ETH/ERC20 (Mint), cToken ERC20 (Redeem)
+     * @param outputAssetA - cToken (Mint), ETH/ERC20 (Redeem)
+     * @param totalInputValue - the amount of ERC20 token/ETH to deposit (Mint), the amount of cToken to burn (Redeem)
+     * @param auxData - 0 (Mint), 1 (Redeem)
+     * @return outputValueA - the amount of cToken (Mint) or ETH/ERC20 (Redeem) transferred to RollupProcessor.sol
```

Comment:
My feeling is that when the variables are not used it looks nicer to just keep them hidden but I am not really sure if it's a good practice. But I think it's cool that it makes it obvious that param is not used when it's not named.

Reply:
Think it can easily be misunderstood as it just being "forgotten" so it is not always clear if used or not. And it going after the natspec with return values, one might expect only a single value so think it is better to give all three in the natspec, even if partially unused.